File: msgraph-mail/ruby/graphrubyv4/utilities/users/users_request_builder.rb
require 'microsoft_kiota_abstractions'
require_relative './item/user_request_builder'

module Graphrubyv4::Utilities::Users
    ## 
    # Builds and executes requests for operations under \users
    class UsersRequestBuilder
        
        ## 
        # Current path for the request
        @current_path
        ## 
        # The http core service to use to execute the requests.
        @http_core
        ## 
        # Path segment to use to build the URL for the current request builder
        @path_segment
        ## 
        # Gets an item from the graphrubyv4.utilities.users collection
        def [](position)
            return UserRequestBuilder.new(@current_path + @path_segment  + "/" + position, @http_core)
        end
        ## 
        ## Instantiates a new UsersRequestBuilder and sets the default values.
        ## @param currentPath Current path for the request
        ## @param httpCore The http core service to use to execute the requests.
        ## @return a void
        ## 
        def initialize(current_path, http_core) 
            @path_segment = "/users"
            @http_core = http_core
            @current_path = current_path
        end
    end
end

File: msgraph-mail/php/src/Users/UsersRequestBuilder.php
<?php

namespace Microsoft\Graph\Users;

use Microsoft\Graph\Users\Item\UserRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

class UsersRequestBuilder 
{
    /** @var array|null $pathParameters Path parameters for the request */
    private array $pathParameters;
    
    /** @var RequestAdapter|null $requestAdapter The request adapter to use to execute the requests. */
    private RequestAdapter $requestAdapter;
    
    /** @var string|null $urlTemplate Url template to use to build the URL for the current request builder */
    private ?string $urlTemplate;
    
    /**
     * Instantiates a new UsersRequestBuilder and sets the default values.
     * @param array $pathParameters Path parameters for the request
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct(array $pathParameters, RequestAdapter $requestAdapter) {
        $this->urlTemplate = 'https://graph.microsoft.com/beta/users';
        $this->requestAdapter = $requestAdapter;
        $this->pathParameters = $pathParameters;
    }

}
